hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/staking/StakingRewardsHandlerImpl.java

```diff
@@ -273,13 +368,22 @@ private void adjustNodeStakes(
         }
     }
 
-    public boolean shouldUpdateStakeStart(
+    /**
+     * List of condition sto validate if the account need to update stakeAtStartOfLastRewardedPeriod.
+     * @param account the account
+     * @param isRewarded if the account is rewarded
+     * @param reward the reward amount
+     * @param stakingRewardStore the staking reward store
+     * @param consensusNow the consensus time
+     * @return true if the account need to update stakeAtStartOfLastRewardedPeriod, false otherwise
+     */
+    public boolean shouldUpdateStakeAtStartOfLastRewardPeriod(
             @Nullable final Account account,
             final boolean isRewarded,
             final long reward,
             @NonNull final ReadableNetworkStakingRewardsStore stakingRewardStore,
             @NonNull final Instant consensusNow) {
-        if (account == null || account.hasStakedAccountId() || account.declineReward()) {
+        if (account == null || !account.hasStakedNodeId() || account.declineReward()) {
```

Comment:
Hmm...until we have https://github.com/hashgraph/pbj/issues/160 from PBJ and can truly "unset" the `oneof staking_id`, doesn't this `!account.hasStakedNodeId()` need to be,
```
account.stakedNodeIdOrElse(SENTINEL_NODE_ID) == SENTINEL_NODE_ID
```
to make sure we behave properly for an account that _was_ staking to a node, but then stopped staking?

Reply:
Ah yes makes sense. Changed